Detectors/ITSMFT/ITS/workflow/src/ThresholdCalibratorSpec.cxx

```diff
@@ -909,52 +928,65 @@ void ITSThresholdCalibrator::addDatabaseEntry(
   char stave[6];
   sprintf(stave, "L%d_%02d", lay, sta);
 
-  o2::dcs::addConfigItem(this->mTuning, "Stave", std::string(stave));
-  o2::dcs::addConfigItem(this->mTuning, "Hs_pos", std::to_string(ssta));
-  o2::dcs::addConfigItem(this->mTuning, "Hic_Pos", std::to_string(mod));
-  o2::dcs::addConfigItem(this->mTuning, "ChipID", std::to_string(chipInMod));
+  if (isQC) {
+    o2::dcs::addConfigItem(this->mChipDoneQc, "Stave", std::string(stave));
+    o2::dcs::addConfigItem(this->mChipDoneQc, "Hs_pos", std::to_string(ssta));
+    o2::dcs::addConfigItem(this->mChipDoneQc, "Hic_Pos", std::to_string(mod));
+    o2::dcs::addConfigItem(this->mChipDoneQc, "ChipID", std::to_string(chipInMod));
+    return;
+  }
+
+  // Get ConfDB id for the chip chipID
+  std::string hs = ssta == 0 ? "L" : "U";
+  int confDBid = mConfDBmap.getConfigDBid(lay < 3 ? Form("L%d_%02d_M%d_C%d", lay, sta, mod, chipInMod) : Form("L%d_%02d%s_M%d_C%d", lay, sta, hs.c_str(), mod, chipInMod));
```

Comment:
In fact, the O2 `chipID` is a contiguous number from 0 to 24119, to map it on some integer confDBid, you don't need to decompose it to ` lay, sta, mod, chipInMod`, then create a string then to query a map, you can simply define a 
`std::array<int, 24120> confDBidMap` with the values you need and query it simply as
`int confDBid = confDBidMap[chipID];`
Still, instead of putting into the code such a huge const array (which I guess may change one day), better to have CCDB entry with such an array and update it when needed.

Reply:
@shahor02 I would stick to your format for the chip location. Anyway now I tried the workflow with the CCDB retrieve of the object containing the map of confDB id. It works (the vector<int> is readout properly) but when running the workflow I see a huge number of messages like:
```
[40890:its-calibrator_0]: [19:13:30][ALARM] Backpressure on channel from_its-stf-decoder_to_its-calibrator_0[0]. Waiting.
[40890:its-calibrator_0]: [19:13:30][INFO] Back to normal on channel from_its-stf-decoder_to_its-calibrator_0[0].
```
At some point the workflow freezes so I cannot fully test locally. The problem is due to the ccdb retrieve of the object because removing it, all the messages above disappear.

Reply:
yes, this is the workflow:
```
o2-raw-tf-reader-workflow --input-data 514354.dat --max-tf -5 --onlyDet ITS --shm-segment-size 40000000000 | \
o2-itsmft-stf-decoder-workflow  --ignore-noise-map --nthreads 1 --pipeline its-stf-decoder:1 --no-clusters --no-cluster-patterns --enable-calib-data --digits --shm-segment-size 40000000000 | \
o2-its-threshold-calib-workflow -b  --meta-output-dir "./" --nthreads 1 --fittype derivative | \
o2-its-threshold-aggregator-workflow -b --ccdb-url "http://ccdb-test.cern.ch:8080"
```
where the 514354.dat contains 3 local TF files.

Reply:
OK, the ITSThresholdCalibrator does not send on every TF while the DPL fetcher sends at every TF and fills the pipeline. 
1st of all, you should have the output types of getITSThresholdCalibratorSpec not 
`Lifetime::Timeframe` but `Lifetime::Sporadic`.
Could you try if this changes something? It might be that one more modification will be needed, but I will need to clarify it with Giulio, yesterday I had a similar situation and it did not work.

Reply:
Ah ok I see, unfortunately even with `Lifetime::Sporadic`, the issue is still there.

Reply:
I see, in principle, https://github.com/shahor02/AliceO2/blob/9d0663a7dc1a34eead2973f6bff1fe4e8ec2a5a0/Detectors/MUON/MCH/Workflow/src/cru-page-reader-workflow.cxx#L493-L508 (with `return broadCastLatest();` issued at every TF, even if the run() does not send output) is supposed to fix such situations, but it did not work and @ktf is checking why.

Note that most probably, in the real EPN workflow, where the aggregator is running on a separate host and does not see the clock tick of every TF, you will not have this 
situation, since the CCDB fetcher will get the object only when the input proxy gets something. But to not rely on that, perhaps at the moment you better comment out ccbParamSpec stuff and load the object via ccdbmanager...

Reply:
ok so you would implement the ccdbmanager solution also for P2 of just for local tests? Yes at P2 I do not expect that as well because we don't see issues with the NOISEMAP in the STFDecoderSpec which uses the ccdbparamspec.

Reply:
STFDecoderSpec is different, it receives both data and CCDB objects  for every TF, so there is no problem. The problem appears when the workflow does not receive data for the TF for which it already got ccdb object: then the workflow gets stuck. 
This is what happening with the aggregator in your local test (because its receiving of the CCDB object is triggered by the 1st device in the workflow, i.e. the reader). This should not happen on the real epn, where the CCDB query will be triggered by the input-proxy only when it receives data. But, I guess you don't want to use different CCDB access mechanisms in the local tests and on the EPN, so I would suggest at the moment to switch to the CCDBManager, unless you can wait a few days: this is a general problem for all aggregators, we (@ktf) should find a solution for it.

Reply:
If data is sporadic, then it should be marked as `Lifetime::Sporadic`. That will guarantee that spurious timeframes are ignored and CCDB data is dropped not matched by detector data will be dropped. If this is not the case, it's a bug and we (I) need to fix it.

Reply:
@ktf Ivan has already tried with `Lifetime::Sporadic` (see discussion above), but it does not work. I guess because the aggregator has both this sporadic input and normal one from the CCDBFetcher.

Reply:
In principle it should have excluded the Lifetime::Condition. I am checking this.

Reply:
@ktf @shahor02 thanks for the explanation. I can wait till Sunday. The idea would be to have this PR for next week at P2.

Reply:
in addition I'm doing the `pc.inputs(..)` of the CCDB object only when the first TF arrives (when `mRunType==-1`). Checking the STFdecoder more in detail I see that this is done for every TF in the run() method. I guess I should do the same. But then the issue with aggregator remains, as you said.